### PR TITLE
[E2E] Fix service quotas for EventBridge rules

### PR DESCRIPTION
**Pull request by Ankitasw**
**What type of PR is this?**
/kind failing-test

**What this PR does / why we need it**:
This PR increases resource quota for EventBridge rules in E2E tests.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [x] adds or updates e2e tests

### CODE REVIEW

File: test/e2e/shared/aws.go [hunk 1]
```diff
@@ -411,6 +411,9 @@ func createCloudFormationStack(prov client.ConfigProvider, t *cfn_bootstrap.Temp
 
 	Eventually(func() bool {
 		err := cfnSvc.ReconcileBootstrapStack(t.Spec.StackName, *renderCustomCloudFormation(t), tags, true)
+		if err != nil {
+			By(fmt.Sprintf("Error reconciling Cloud formation stack %v", err))
```
Comment (Skarlso) [inline test/e2e/shared/aws.go:415]: 👍

**Comment by Ankitasw** [comment]
I have seen these errors in release-1.5 pipelines, so

/cherry-pick release-1.5

**Comment by Skarlso** [comment]
@Ankitasw What do you think in adding this thing? 
https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4053/files

We can, however, look for a fix on that CloudFormation reconcile loop in a next PR. :)

**Comment by Ankitasw** [comment]
@Skarlso yeah was thinking to handle it separately. Once both these tests passes, it would be good to merge.

**Comment by Skarlso** [comment]
>@Skarlso yeah was thinking to handle it separately. Once both these tests passes, it would be good to merge.
Okay. :) Sounds reasonable. :)

**Comment by Ankitasw** [comment]
@Skarlso PR ready for review now.

**Comment by Ankitasw** [comment]
There are some merge conflicts, I will open a separate PR for release-1.5

**Comment by Ankitasw** [comment]
Rethinking, as accounts might be same for both the releases 😅 so will wait for few days.

**Pull request merged**